posthog/src/main/java/com/posthog/android/PostHog.java

```diff
@@ -797,6 +803,18 @@ void enqueue(BasePayload payload) {
     chain.proceed(payload);
   }
 
+  void getSessionId() {
+    Properties properties = propertiesCache.get();
+    Persistence persistence = persistenceCache.get();
+    String sessionId = properties.sessionId();
+    Instant sessionLastTimestamp = persistence.sessionLastTimestamp();
+    if (sessionId == null || (Duration.between(sessionLastTimestamp, Instant.now()).getSeconds() > this.sessionExpirationTimeSeconds)) {
+      String newSessionId = UUID.randomUUID().toString();
+      properties.putSessionId(newSessionId);
+    }
+    persistence.putSessionLastTimestamp(Instant.now());
```

Comment:
Does Properties persist the sessionId? I'm a bit unclear how this works

Reply:
bump the min sdk version to 26 to use java's time utils. We already made the change to use java 8 in the previous PR

Reply:
Yes there's a Properties and a Persistence for caching. I think properties is meant for values that get saved into properties that's why it holds sessionId but not sessionTimestamp